pkg/kubernetes/selinux.go

```diff
@@ -0,0 +1,82 @@
+package kubernetes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/suse-edge/edge-image-builder/pkg/http"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"go.uber.org/zap"
+)
+
+const (
+	rke2SELinuxPolicyReleaseURL = "https://github.com/rancher/rke2-selinux/releases/download/%s/%s"
+	k3sSELinuxPolicyReleaseURL  = "https://github.com/k3s-io/k3s-selinux/releases/download/%s/%s"
+
+	rancherSigningKeyURL = "https://rpm.rancher.io/public.key"
+)
+
+type selinuxPolicy struct {
+	downloadURL string
+	rpmName     string
+}
+
+func DownloadSELinuxRPMs(kubernetes *image.Kubernetes, rpmDir, gpgKeysDir string) error {
+	policy := newSELinuxPolicy(kubernetes.Version)
+	policyPath := filepath.Join(rpmDir, policy.rpmName)
+
+	if _, err := os.Stat(policyPath); err != nil {
+		if !errors.Is(err, fs.ErrNotExist) {
+			zap.S().Errorf("Looking for SELinux RPM '%s' failed unexpectedly: %s", policyPath, err)
+		}
+
+		if err = http.DownloadFile(context.Background(), policy.downloadURL, policyPath, nil); err != nil {
+			return fmt.Errorf("downloading selinux policy: %w", err)
+		}
+	}
+
+	signingKeyPath := filepath.Join(gpgKeysDir, "rancher-public.key")
+	if err := http.DownloadFile(context.Background(), rancherSigningKeyURL, signingKeyPath, nil); err != nil {
+		return fmt.Errorf("downloading rancher signing key: %w", err)
+	}
+
+	return nil
+}
+
+func newSELinuxPolicy(kubernetesVersion string) selinuxPolicy {
+	const (
+		rke2PolicyVersion       = "0.17"
+		rke2PolicyChannel       = "stable"
+		rke2PolicyReleaseNumber = "1"
+
+		k3sPolicyVersion       = "1.4" // TODO: Bump to 1.5 once the release assets are fixed
+		k3sPolicyChannel       = "stable"
+		k3sPolicyReleaseNumber = "1"
+
+		rke2PolicyRPM = "rke2-selinux-%s-%s.slemicro.noarch.rpm"
+		k3sPolicyRPM  = "k3s-selinux-%s-%s.slemicro.noarch.rpm"
+	)
```

Comment:
I understand the approach, but we're going to have to think about version management here. We will have branches of EIB for the respective SUSE Edge version, but it's still something for us to think about.

Reply:
You're absolutely correct. We'll need to figure out a way to extract separate configuration values. At this point though, hardcoding these is our best bet to get some kind of automation in right now.